### ISSUE TITLE
tfd.Categorical log_prob returns `nan` in XLA when any `prob` is `0`

### ISSUE THREAD

**Issue by chrism0dwk** [tensorflow/probability#975]
With **tfp-nightly** (as of the date of this post) and XLA-compilation, `tfd.Categorical.log_prob()` appears to return `nan` if at least one element of `logits` is `-inf`.  If run in regular graph mode, a floating point number is returned as expected. 

[Minimal example](https://colab.research.google.com/drive/1k1n-oRsCnkm28cB4W1KDu6noXB0mx-cF?usp=sharing) 
```
probs = tf.constant([1, 0, 1], tf.float32)
logits = tf.math.log(probs)

@tf.function(experimental_compile=True)
def f():
  X = tfd.Categorical(logits=logits)
  x = X.sample()
  return X.log_prob(x)

print(f())  # Returns `nan`
```

If run without XLA compilation, the correct answer is obtained.

**Comment by chrism0dwk** [comment]
This turns out to be a more fundamental issue with `tf.nn.sparse_softmax_cross_entropy_with_logits` as mentioned [here](https://github.com/tensorflow/tensorflow/issues/40606).

**Comment by brianwa84** [comment]
@axch did we fix this in TFP? I recall you fixing something to do with inf logits.

**Comment by srvasude** [comment]
This is fixed by https://github.com/tensorflow/tensorflow/commit/e9b3d33ccfa4b670ca0a221db9bc7bc1281b0557
